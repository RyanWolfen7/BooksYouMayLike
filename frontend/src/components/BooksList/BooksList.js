import React, { useState } from 'react'
import { Paper, Grid } from '@material-ui/core'
import { BookListStyles } from '../../styles/BooksListStyles'

const BooksList = props => {
    const classes = BookListStyles()
    // fake for purpose of styling remove later
    const [ booksList, setBooksList ] = useState(['a','b','c','d'])

    const renderBookCards = () => {
        return booksList.map( book => {
            return (
                <Grid item xs={3} sm={3}>
                    <Paper className={classes.paper}> This is a Book </Paper>
                </Grid>
            )
        })
    }

    return (
    <div className={classes.root}>
        <Grid container spacing={3}>
            {renderBookCards()}
        </Grid>
    </div>
    )
}

export default BooksList