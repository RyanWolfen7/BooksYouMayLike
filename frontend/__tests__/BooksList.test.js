import React from 'react'
import { shallow } from 'enzyme'
import BooksList from '../src/components/BooksList/BooksList'

let List

beforeAll(() => {
    List = shallow(<BooksList />)
})

test('List to match snapshot', () => {
    expect(List).toMatchSnapshot()
})