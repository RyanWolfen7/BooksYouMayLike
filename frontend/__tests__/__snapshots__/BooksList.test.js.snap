// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List to match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={3}
      xs={3}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
         This is a Book 
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={3}
      xs={3}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
         This is a Book 
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={3}
      xs={3}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
         This is a Book 
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={3}
      xs={3}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
         This is a Book 
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
